source/Calamari.Tests/Fixtures/PackageRetention/LeastFrequentlyUsedWithAgingCacheAlgorithmFixture.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Calamari.Common.Plumbing.Deployment.PackageRetention;
+using Calamari.Deployment.PackageRetention.Caching;
+using Calamari.Deployment.PackageRetention.Model;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Calamari.Tests.Fixtures.PackageRetention
+{
+    [TestFixture]
+    public class LeastFrequentlyUsedWithAgingCacheAlgorithmFixture
+    {
+        [Test]
+        public void WhenSpaceNeededIsGreaterThanSpaceUsed_ThenThrowException()
+        {
+            var lfu = new LeastFrequentlyUsedWithAgingCacheAlgorithm();
+            Assert.Throws<InsufficientCacheSpaceException>(() => lfu.GetPackagesToRemove(new List<JournalEntry>(), 10_000));
+        }
+
+        [Test]
+        public void WhenPackageIsLocked_ThenDoNotConsiderItForRemoval()
+        {
+            var lfu = new LeastFrequentlyUsedWithAgingCacheAlgorithm();
+            var entry = CreateEntry("package-1", "1.0", 20, ("task-1", 2));
+            entry.AddLock(new ServerTaskId("task-0"), new CacheAge(1));
+
+            Assert.Throws<InsufficientCacheSpaceException>(() => lfu.GetPackagesToRemove(new List<JournalEntry>(), 10));
+        }
+
+        [Test]
+        public void WhenUsingAllThreeFactors_ThenReturnThePackageWithTheLowestValue()
+        {
+            var spaceNeeded = 10;
+
+            var entries = new[]
+            {
+                CreateEntry("package-1", "1.0", 10, ("task-1", 2), ("task-2", 3)),
+                CreateEntry("package-2", "1.1", 10, ("task-3", 1)),
+                CreateEntry("package-3", "1.0", 10, ("task-4", 2)), //Lower value - has a newer version, and is only used once.
+                CreateEntry("package-3", "1.1", 10, ("task-5", 3))
+            };
+
+            var packagesToRemove = new LeastFrequentlyUsedWithAgingCacheAlgorithm(0.5M, 1, 1)
+                                   .GetPackagesToRemove(entries, spaceNeeded)
+                                   .OrderBy(p => p.PackageId.Value)
+                                   .ThenBy(p => p.Version)
+                                   .ToList();
+
+            packagesToRemove.Should()
```

Comment:
I think the assertion can be simplified to:
```
packagesToRemove.Should().BeEquivalentTo(new PackageIdentity("package-3", "1.0"));
```

Reply:
That's just _slightly_ better 😅